### ISSUE TITLE
Support paying amountless bolt11 

### ISSUE THREAD

**Issue by thesimplekid** [cashubtc/cdk#460]
https://github.com/cashubtc/nuts/pull/173

**Comment by mubarak23** [comment]
@thesimplekid let me have take at this

**Comment by mubarak23** [comment]
when a mint `PostMeltQuoteBolt11Response` has amount set to 0, this mean the wallet is trying to get an amountless (zero-amount invoice) request invoice  in the request a melt quote,

when this happen, we are going to check mint `MeltMethodSetting` if it's support for amountless bolt11

@thesimplekid

**Comment by thesimplekid** [comment]
No, the `PostMeltQuoteBolt11Request` should be extend to have an optional field as in the linked nut pr of the amount in the case of an amount less invoice. If this is supported by the mint then that amount should be paid. There should also be a check that if there is also an amount in the invoice it is the same as requested. The `PostMeltQuoteBolt11Response` does not need to change.

**Comment by mubarak23** [comment]
i understand, so if there is an option amount field set in  `PostMeltQuoteBolt11Request` then mint should check if it's support `amountless` (zero-amount invoice), then ensure the amount field match what is in the request bolt11 Lightning invoice

**Comment by mubarak23** [comment]
> Yes.
> 
> > then ensure the amount field match what is in the request bolt11 Lightning invoice
> 
> Only if there is amount included in the invoice.

can an invoice comes without an amount
